pg-index-health-jdbc-connection/src/test/java/io/github/mfvanek/pg/connection/PgConnectionImplTest.java

```diff
@@ -92,8 +92,18 @@ void equalsHashCodeShouldAdhereContracts() {
 
     @Test
     void toStringTest() {
-        final PgConnection connection = PgConnectionImpl.ofPrimary(getDataSource());
+        final PgConnection connection = PgConnectionImpl.of(getDataSource(), PgHostImpl.ofUrl("jdbc:postgresql://primary:5432"));
         assertThat(connection)
-                .hasToString("PgConnectionImpl{host=PgHostImpl{pgUrl='jdbc:postgresql://primary', hostNames=[primary], maybePrimary=true}}");
+                .hasToString("PgConnectionImpl{host=PgHostImpl{pgUrl='jdbc:postgresql://primary:5432', hostName=primary, port=5432, maybePrimary=true}}");
+    }
+
+    @Test
+    void twoConnectionsDifferentSameHostWithDifferentPortsConsideredNotEqual() {
+        final DataSource firstDatasource = Mockito.mock(DataSource.class);
+        final DataSource secondDatasource = Mockito.mock(DataSource.class);
+        final PgConnection firstPgConnection = PgConnectionImpl.of(firstDatasource, PgHostImpl.ofUrl("jdbc:postgresql://localhost:5432"));
```

Comment:
`DataSource `should be the same

Reply:
It seems a little confusing to me. But I think I got it.